opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDataType.java

```diff
@@ -97,56 +98,86 @@ public ExprType getExprType() {
             instances.put(t.toString(), OpenSearchDataType.of(t)));
   }
 
+  /**
+   * Parses index mapping and maps it to a Data type in the SQL plugin.
+   * @param indexMapping An input with keys and objects that need to be mapped to a data type.
+   * @return The mapping.
+   */
+  public static Map<String, OpenSearchDataType> parseMapping(Map<String, Object> indexMapping) {
+    Map<String, OpenSearchDataType> result = new LinkedHashMap<>();
+
+    if (indexMapping == null) {
+      return result;
+    }
+
+    indexMapping.forEach((k, v) -> {
+      var innerMap = (Map<String, Object>)v;
+      // by default, the type is treated as an Object if "type" is not provided
+      var type = ((String) innerMap
+          .getOrDefault(
+              "type",
+              "object"))
+          .replace("_", "");
+      if (!EnumUtils.isValidEnumIgnoreCase(OpenSearchDataType.MappingType.class, type)) {
+        // unknown type, e.g. `alias`
+        // TODO resolve alias reference
+        return;
+      }
+      // create OpenSearchDataType
+      result.put(k, OpenSearchDataType.of(
+          EnumUtils.getEnumIgnoreCase(OpenSearchDataType.MappingType.class, type),
+          innerMap)
+      );
+    });
+    return result;
+  }
+
   /**
    * A constructor function which builds proper `OpenSearchDataType` for given mapping `Type`.
    * @param mappingType A mapping type.
    * @return An instance or inheritor of `OpenSearchDataType`.
    */
-  public static OpenSearchDataType of(MappingType mappingType) {
-    var res = instances.getOrDefault(mappingType.toString(), null);
-    if (res != null) {
-      return res;
-    }
-    ExprCoreType exprCoreType = mappingType.getExprCoreType();
-    if (exprCoreType == ExprCoreType.UNKNOWN) {
-      switch (mappingType) {
+  public static OpenSearchDataType of(MappingType mappingType, Map<String, Object> innerMap) {
+    OpenSearchDataType res = instances.getOrDefault(mappingType.toString(),
+        new OpenSearchDataType(mappingType)
```

Comment:
Why not `null` here? I think `instances` should contain all default instances since the very beginning.

Reply:
Changed in https://github.com/Bit-Quill/opensearch-project-sql/pull/268/commits/58a8f193ffe1407665734d55a60e53ff9bc97aa8